src/Parsers/ASTCreateQuery.h

```diff
@@ -112,7 +112,7 @@ class ASTCreateQuery : public ASTQueryWithTableAndOutput, public ASTQueryWithOnC
     String as_table;
     IAST * as_table_function = nullptr;
     ASTSelectWithUnionQuery * select = nullptr;
-    IAST * comment = nullptr;
+    ASTPtr comment;
```

Comment:
I think it is not necessary and moreover is harmful. `IAST` abstraction assumes that all ast fields must be in children array to allow uniform traversal of ast. That also implies that if you update an ast field you must update it in `children` array. `IAST:set` and `IAST:reset` handle this. Unfortunately is not strictly forced and is violated sometimes.

Reply:
Understood. That simplifies *a lot* the changes needed for this: https://github.com/ClickHouse/ClickHouse/pull/62117/commits/9d87cec8433442af9f560c3d1d72f0d3e01e0c4f

Thanks for the insightful feedback :)